safety_shield/src/safety_shield.cc

```diff
@@ -452,60 +478,181 @@ void SafetyShield::computesPotentialTrajectory(bool v, const std::vector<double>
-            if(!is_under_iso_velocity_) {
-                *under_vel_motion = long_term_trajectory_.interpolate(vel_s_d, vel_ds_d, vel_dds_d, ddds_d, v_max_allowed_, a_max_allowed_, j_max_allowed_);
+            spdlog::error("Error in computesPotentialTrajectoryForPFL: no failsafe-path is set when planning repair path?");
+        }
+
+        // plan repair path and replace
+        recovery_path_correct_ = planSafetyShield(pos, vel, acc, 1, a_max_manoeuvre, j_max_manoeuvre, recovery_path_);
+    }
+    // Only plan new failsafe trajectory if the recovery path planning was successful.
+    if (recovery_path_correct_) {
+        //advance one step on repair path
+        recovery_path_.increment(sample_time_);
+
+        double s_d, ds_d, dds_d;
+        // Calculate goal
+        is_under_iso_velocity_ = false;
+        // v_max is maximum of LTT or STP and vel_s_dot is how much path velocity needs to be scaled to be under v_iso
+        failsafe_path_2_pfl_.getFinalMotion(s_d, ds_d, dds_d);
+        double v_max;
+        if(safety_method_ == LTT_MAXIMUM && !new_ltt_) {
+            v_max = long_term_trajectory_.getMaxofMaximumCartesianVelocity();
+        } else if(safety_method_ == LTT_MAXIMUM && new_ltt_) {
+            v_max = new_long_term_trajectory_.getMaxofMaximumCartesianVelocity();
+        } else if(safety_method_ == STP_MAXIMUM && !new_ltt_) {
+            v_max = long_term_trajectory_.getMaxofMaximumCartesianVelocityWithS(s_d);
+        } else {
+            // safety_method_ == STP_MAXIMUM && new_ltt_
+            v_max = new_long_term_trajectory_.getMaxofMaximumCartesianVelocityWithS(s_d);
+        }
+        is_under_iso_velocity_ = v_max <= v_iso_;
+        double v_limit;
+        bool failsafe_2_pfl_planning_success = true;
+        // TODO: potential mistake: if its under iso-velocity, we dont need to compute the failsafe-path of the PFL-criterion and maybe I do it wrong
+        if(!is_under_iso_velocity_) {
+            v_limit = v_iso_ / v_max;
+            if(v_limit >= 1 || v_limit <= 0) {
+                spdlog::error("v_limit not between 0 and 1");
             }
+            failsafe_2_pfl_planning_success = planSafetyShield(recovery_path_.getPosition(), recovery_path_.getVelocity(), recovery_path_.getAcceleration(), v_limit, a_max_manoeuvre, j_max_manoeuvre, failsafe_path_2_pfl_);
         }
-        goal_motion->setTime(potential_path_.getPhase(3));
+
+        // plan new failsafe path for STP
+        bool failsafe_2_static_planning_success = planSafetyShield(recovery_path_.getPosition(), recovery_path_.getVelocity(), recovery_path_.getAcceleration(), 0, a_max_manoeuvre, j_max_manoeuvre, failsafe_path_2_static_);
+
+        // Check the validity of the planned path
+        if (!failsafe_2_static_planning_success || !failsafe_2_pfl_planning_success || recovery_path_.getPosition() < failsafe_path_static_.getPosition() || recovery_path_.getPosition() < failsafe_path_pfl_.getPosition()) {
```

Comment:
This is what causes your error. 
It can happen that
```
recovery_path_.getPosition() < failsafe_path_pfl_.getPosition()
```
if the robot is braking along the static failsafe.